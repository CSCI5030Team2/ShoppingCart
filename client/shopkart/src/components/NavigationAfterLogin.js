import React, { Component } from "react";
import { Link } from "react-router-dom";
// import DisplayProducts from "./DisplayProducts";
import { getProducts,AddtoCart } from "../actions/products";
import { connect } from "react-redux";
// import logo from "../logo.PNG";
import LoginNavbar from "./LoginNavbar";
import axios from "axios";
import AdsHolder from "./AdsHolder";

export class NavigationAfterLogin extends Component {
  componentDidMount() {
    this.props.getProducts();
  }



  // componentWillMount() {
  //   if (!localStorage.getItem("token")) {
  //     this.props.history.push("/");
  //   }
  // }

  
  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

    
  render() {
    //console.log("NavigationAfterLogin "+this.props.getProducts())
    return (
      <div>
        <div>
          <LoginNavbar />
        </div>
        <h2 style={{ textAlign: "center", marginTop: 2 + "em" }}>Products</h2>
        <div id="outerDiv">
           <AdsHolder/>
            {
                this.props.products.map(product => (
                    <div>
                      <div
                        className="UserProductDisplay"
                        style={{ width: 15 + "em" }}
                      >
                        <p>
                          <b>Product Name : </b> {product.itemName}
                        </p>
                        <p>
                          <b> Quantity : </b> {product.quantity}
                        </p>
                        <p>
                          <b>Price : </b> ${product.price}
                        </p>
                        <button
                          onClick={() => {
                              //console.log(localStorage.getItem("token"))
                              //console.log(product.itemName)

                              //temp solution, this does not use product.js AddtoCart implementation
                              axios.post("http://localhost:8080/cart",
                                  {
                                      token: localStorage.getItem("token"),
                                      itemName: product.itemName,
                                      quantity: 1
                                  }

                                  //returned msg is logged for dev confirmation purpose
                              ).then(r => console.log(r.data))

                              //this is ugly, try to beautify it a bit
                              alert("Saved 1 " + product.itemName +" to cart")


                          }}
                          id="editBtn"
                        >
                          Add To Cart
                        </button>
                      </div>
                    </div>
          ))}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  products: state.productReducer.products
});

export default connect(
  mapStateToProps,
  { getProducts,AddtoCart }
)(NavigationAfterLogin);
