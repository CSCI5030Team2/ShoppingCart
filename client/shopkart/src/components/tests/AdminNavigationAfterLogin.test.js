import React from "react";
import { mount } from "enzyme";
import NavigationAfterLogin from "../NavigationAfterLogin";
import { Provider } from "react-redux";
import store from "../../store";
import { BrowserRouter } from "react-router-dom";

const Navigationafterlogin = jest.fn();
const wrapper = mount(
  <Provider store={store}>
  <BrowserRouter>
    <NavigationAfterLogin Navigationafterlogin={Navigationafterlogin} />
    </BrowserRouter>
  </Provider>
);

describe("Test NavigationAfterLogin Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have exactly one h2 tag", () => {
    expect(wrapper.find("h2").length).toBe(1);
  });

  it("should have exactly four div tags", () => {
    expect(wrapper.find("div").length).toBe(6);
  });
});
