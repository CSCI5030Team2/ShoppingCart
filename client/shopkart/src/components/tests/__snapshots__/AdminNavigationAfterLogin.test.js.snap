// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavigationAfterLogin Component should render the component 1`] = `ReactWrapper {}`;
